soccer-records-model/src/main/java/cz/muni/fi/pa165/team/TeamMatchFacade.java

```diff
@@ -0,0 +1,170 @@
+package cz.muni.fi.pa165.team;
+
+import javax.persistence.EntityManager;
+import java.util.ArrayList;
+import java.util.Date;
+import java.util.List;
+import java.util.UUID;
+
+/**
+ * @author Tomas Smid <smid.thomas@gmail.com>
+ */
+public class TeamMatchFacade
+{
+
+    private TeamMatchService teamMatchService;
+
+    private TeamMatchRepository teamMatchRepository;
+
+    private TeamMatchGoalRepository teamMatchGoalRepository;
+
+    private TeamRepository teamRepository;
+
+    private TeamPlayerRepository teamPlayerRepository;
+
+    private EntityManager entityManager;
+
+    public TeamMatchFacade(
+        TeamMatchService teamMatchService,
+        TeamMatchRepository teamMatchRepository,
+        TeamMatchGoalRepository teamMatchGoalRepository,
+        TeamRepository teamRepository,
+        TeamPlayerRepository teamPlayerRepository,
+        EntityManager entityManager
+    )
+    {
+        this.teamMatchService = teamMatchService;
+        this.teamMatchRepository = teamMatchRepository;
+        this.teamMatchGoalRepository = teamMatchGoalRepository;
+        this.teamRepository = teamRepository;
+        this.teamPlayerRepository = teamPlayerRepository;
+        this.entityManager = entityManager;
+    }
+
+    /**
+     * This method creates and stored a match in DB.
+     *
+     * @param homeTeamId id of home team in the match
+     * @param awayTeamId id of away team in the match
+     * @param startTime start time of the match
+     * @param endTime end time of the match, can be null
+     * @return newly created match
+     */
+    public TeamMatch createMatch(UUID homeTeamId, UUID awayTeamId, Date startTime, Date endTime)
+    {
+        Team homeTeam = teamRepository.getTeamById(homeTeamId);
+        Team awayTeam = teamRepository.getTeamById(awayTeamId);
+        List<TeamMatch> allMatchesOfHomeTeam =
+            new ArrayList<>(teamMatchRepository.findAllMatchesOfTeam(homeTeam.getId()));
```

Comment:
tady jsme řikali, že nějak profiltrujem na podobný zápasy, aby to nebylo tak silene neefektivni

Úplně nejlíp napsat dotaz, kterej hledá fakt jeden specifickej zapas, jestli tam už není, stejně jako když hledám usera podle emailu a předávat jenom ten. Ale kolekce "podobných" taky stačí.